File: src/api/discussion-delete.ts
import { Discussion, DiscussionCommand, SessionHandle } from "~/models";
import { discussions } from "./discussions";
import { discussionPostCommand } from "./private/discussion-post-command";

export const discussionDelete = async (session: SessionHandle, discussion: Discussion): Promise<void> => {
  await discussionPostCommand(session, DiscussionCommand.Delete, {
    possessions: discussion.possessions
  });

  // Will also remove the reference in the cache.
  await discussions(session, discussion.cache);
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/discussion-delete.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.private.discussionPostCommand
import ink.literate.pawnote.models.Discussion
import ink.literate.pawnote.models.DiscussionCommand
import ink.literate.pawnote.models.SessionHandle
import kotlinx.serialization.json.buildJsonObject

suspend fun discussionDelete (session: SessionHandle, discussion: Discussion) {
    discussionPostCommand(session, DiscussionCommand.Delete, buildJsonObject {
        put("possessions", discussion.possessions)
    })

    // Will also remove the reference in the cache.
    discussions(session, discussion.cache)
}